MDANSE/Src/MDANSE/Mathematics/Signal.py

```diff
@@ -292,3 +300,757 @@ def get_spectrum(signal, window=None, timeStep=1.0, axis=0, fft="fft"):
+
+    @property
+    def coeffs(self) -> TransferFunction:
+        """Returns the filter's transfer function numerator and denominator coefficients.
+
+        :Returns:
+            #. TransferFunction: Coefficients for the filter transfer function
+        """
+        return self._coeffs
+
+    @coeffs.setter
+    def coeffs(self, expr: TransferFunction) -> None:
+        """Sets the filter's transfer function numerator and denominator coefficients.
+
+        :Parameters:
+            #. expr (np.array): the rational polynomial expression for the filter transfer function, in terms of its numerator and denominator coefficients
+        """
+        self._coeffs = expr
+
+    @property
+    def nyquist(self) -> float:
+        """Returns the nyquist limit for the filter sample frequency.
+
+        :Returns:
+            #. float: Nyquist limit
+        """
+        return self.sample_freq / 2
+
+    @property
+    def custom_freq_range(self) -> np.ndarray:
+        """Returns the custom frequency range for use with filter plotting
+
+        :Returns:
+            #. np.ndarray: Coefficients for the filter transfer function
+        """
+        return self._custom_freq_range
+
+    @custom_freq_range.setter
+    def custom_freq_range(self, range: np.ndarray) -> None:
+        """Set the custom frequency range for use with filter plotting
+
+        :Parameters:
+            #. range (np.ndarray): Array of values specifying the frequency range
+        """
+        self._custom_freq_range = range
+
+    @staticmethod
+    def frequency_range(
+        N: int, timestep: float, resize_to: int = 1000, symmetric: bool = False
+    ) -> np.ndarray:
+        """Obtain an FFT-based frequency range for the frequency domain of a discrete time signal with a given number of elements and a constant time step.
+
+        :Parameters:
+            #. N (int): Number of samples in input signal (to which filter will be applied to)
+            #. timestep (float): Input signal timestep in picoseconds
+            #. resize_to (int): Up- or down- sample the frequency range array to a given length
+            #. symmetric (bool): If true, retain symmetric property of frequencies, else take only one half of the frequencies
+        :Returns:
+            #. np.ndarray: Symbolic polynomial string
+        """
+        axis_frequencies = fftpack.fftfreq(N, timestep)
+        limit = np.int32(np.floor(len(axis_frequencies) / 2)) if not symmetric else -1
+        return np.linspace(axis_frequencies[0], axis_frequencies[limit], resize_to)
+
+    def set_filter_attributes(self, attributes: dict) -> None:
+        """Update filter instance attributes
+
+        :Parameters:
+            #. attributes (dict): Dictionary containing filter attributes
+        """
+        settings = self.default_settings
+
+        for attr in settings.keys():
+            self.__dict__.update({attr: attributes.get(attr, settings[attr]["value"])})
```

Comment:
```suggestion
            setattr(self, attr, attributes.get(attr, settings[attr]["value"]))
```
 though this would probably be better as:
```
for key, default in settings.items():
    setattr(self, key, attributes.get(key, default["value"]))
```